tests/Secondaries_TEST.cxx

```diff
@@ -154,6 +154,47 @@ TEST(SecondaryVector, EntryPointExitPointRePropagation)
     EXPECT_TRUE(exit_point->propagated_distance == sphere.GetPosition().GetZ() + sphere.GetRadius());
 }
 
+
+TEST(SecondaryVector, HitDetector) {
+    // define our dummy particle track
+    Secondaries dummy_track(nullptr, std::vector<Sector>{});
+    std::vector<Cartesian3D> positions{ {0, 0, 0}, {0, 0, 10}, {0, 0, 20} };
+
+    for (auto p : positions) {
+        ParticleState p_state;
+        p_state.position = p;
+        p_state.direction = Cartesian3D(0, 0, 1);
+        dummy_track.push_back(p_state, InteractionType::Undefined);
+    }
+
+    // test geometries along propagation axis
```

Comment:
Some test spheres seem redundant, while I would also suggest testing the other geometries.
Would be interesting to see, if the edge case of propagating at the surface of a box is also handled as expected.

Reply:
I added some tests for other geometries, although I think that this is rather something that should be tested in the UnitTests of the corresponding geometries.
However, I still think that the border case that you described could potentially be pretty important.